File: 0606_Construct String from Binary Tree/ConstructStringfromBinaryTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public String tree2str(TreeNode t) {
        if(t == null) return "";
        StringBuffer sb = new StringBuffer();
        sb.append(t.val);
        
        if(t.left!=null){
            sb.append("(" + tree2str(t.left)+")");
        }

        if(t.right!=null){
            if(t.left==null) sb.append("()");
            sb.append("(" + tree2str(t.right)+")");
        }
        return sb.toString();       
    }
}

File: 0606_Construct String from Binary Tree/ConstructStringfromBinaryTree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public string Tree2str(TreeNode t) {
        if(t == null) return "";
        var sb = new StringBuilder();
        sb.Append(t.val);
        
        if(t.left!=null){
            sb.Append("(" + Tree2str(t.left)+")");
        }

        if(t.right!=null){
            if(t.left==null) sb.Append("()");
            sb.Append("(" + Tree2str(t.right)+")");
        }
        return sb.ToString();
    }
}